todoman/ui.py

```diff
@@ -251,9 +251,23 @@ def detailed(self, todo):
         return rv
 
     def format_date(self, date):
+        """
+        Returns date in the following format:
+        * if date == today or tomorrow: "Today" or "Tomorrow"
+        * else: return a string representing that date
+        * if no date is supplied, it returns empty_date
+        """
         if date:
-            rv = date.strftime(self.date_format)
-            return rv
+            date_tomorrow = self.now.date() + timedelta(days=1)
+            # Get a dictionary which stores the dates for which
+            # we don't have to merely return a date string and
+            # map it to the special string we need to return
+            WIDTH = len(date.strftime(self.date_format))
```

Comment:
Actually, `len(empty_date)` was fine. They should both have the same with, and we avoid formatting this date when it's [possibly] unnecessary.

I'd also prefer move this over to `__init__` , so we only calculate  this width once, instead every time we format a date (pretty much like we only calculate `empty_date` once.

Let me know if you're not very familiar with contructors/`__init__` or if this is otherwise unclear.

Also, `date_width` is actually a better variable name, but that one's my fault because I used `WIDTH` on my example, which really was a bad idea on my part, sorry.